apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/FissClaimTransformerV2.java

```diff
@@ -0,0 +1,229 @@
+    return new Reference()
+        .setIdentifier(
+            new Identifier()
+                .setType(
+                    new CodeableConcept()
+                        .setCoding(
+                            Collections.singletonList(
+                                new Coding(
+                                    "http://terminology.hl7.org/CodeSystem/v2-0203",
+                                    "MC",
+                                    "Patient's Medicare number"))))
+                .setSystem("http://hl7.org/fhir/sid/us-mbi")
+                .setValue(claimGroup.getMbi()));
+  }
+
+  private static List<Claim.DiagnosisComponent> getDiagnosis(PreAdjFissClaim claimGroup) {
+    return Arrays.asList(
+        new Claim.DiagnosisComponent()
+            .setSequence(1)
+            .setDiagnosis(
+                new CodeableConcept()
+                    .setCoding(
+                        Collections.singletonList(
+                            new Coding(
+                                "http://hl7.org/fhir/sid/icd-10",
+                                claimGroup.getPrincipleDiag(),
+                                null))))
+            .setType(
+                Collections.singletonList(
+                    new CodeableConcept()
+                        .setCoding(
+                            Collections.singletonList(
+                                new Coding(
+                                    "http://terminology.hl7.org/CodeSystem/ex-diagnosistype",
+                                    "principal",
+                                    "Principal Diagnosis"))))),
+        new Claim.DiagnosisComponent()
+            .setSequence(2)
+            .setDiagnosis(
+                new CodeableConcept()
+                    .setCoding(
+                        Collections.singletonList(
+                            new Coding(
+                                "http://hl7.org/fhir/sid/icd-10",
+                                claimGroup.getAdmitDiagCode(),
+                                null))))
+            .setType(
+                Collections.singletonList(
+                    new CodeableConcept()
+                        .setCoding(
+                            Collections.singletonList(
+                                new Coding(
+                                    "http://terminology.hl7.org/CodeSystem/ex-diagnosistype",
+                                    "admitting",
+                                    "Admitting Diagnosis"))))));
+  }
+
+  private static List<Claim.ProcedureComponent> getProcedure(PreAdjFissClaim claimGroup) {
+    List<Claim.ProcedureComponent> procedure = new ArrayList<>();
+
+    // Sort proc codes by priority prior to building resource
+    List<PreAdjFissProcCode> procCodes = new ArrayList<>(claimGroup.getProcCodes());
+    procCodes.sort(Comparator.comparingInt(PreAdjFissProcCode::getPriority));
+
+    int i = 0;
+
+    for (PreAdjFissProcCode procCode : procCodes) {
```

Comment:
minor; but if we need to initialize an index for the sequence anyways, would the following be preferable?
```
for(int i=0;  i<procCodes.size(); i++)
```